### PR TITLE
Updating driverless

### PR DESCRIPTION
**Pull request by nidhijainpnp**
Testing results:
```
nidhi@nidhi-X510UNR:~$ driverless _ipps._tcp
ipps://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipps._tcp.local/
ipps://hello1._ipps._tcp.local/

nidhi@nidhi-X510UNR:~$ driverless _ipp._tcp
ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D%20(USB%201)._ipp._tcp.local/
ipp://hello1._ipp._tcp.local/
ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipp._tcp.local/

nidhi@nidhi-X510UNR:~$ driverless 
ipps://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipps._tcp.local/
ipps://hello1._ipps._tcp.local/
ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D%20(USB%201)._ipp._tcp.local/

nidhi@nidhi-X510UNR:~$ driverless list
DEBUG: Started ippfind (PID 24180)
"driverless:ipps://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipps._tcp.local/" en "HP" "HP OfficeJet Pro 6960, fully driverless, cups-filters 1.27.5" "MFG:HP;MDL:OfficeJet Pro 6960;CMD:PCLM,PCL,PWGRaster,AppleRaster,JPEG,URF,PWG;"
"driverless:ipps://hello1._ipps._tcp.local/" en "Example" "Example Printer, fully driverless, cups-filters 1.27.5" "MFG:Example;MDL:Printer;CMD:PWGRaster,AppleRaster,PWG,URF;"
DEBUG: Started ippfind (PID 24184)
"driverless:ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipp._tcp.local/" en "HP" "HP OfficeJet Pro 6960, fully driverless, cups-filters 1.27.5" "MFG:HP;MDL:OfficeJet Pro 6960;CMD:PCLM,PCL,PWGRaster,AppleRaster,JPEG,URF,PWG;"
"driverless:ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D%20(USB%201)._ipp._tcp.local/" en "HP" "HP OfficeJet Pro 6960, fully driverless, cups-filters 1.27.5" "MFG:HP;MDL:OfficeJet Pro 6960;CMD:PCLM,PCL,PWGRaster,AppleRaster,JPEG,URF,PWG;"
"driverless:ipp://hello1._ipp._tcp.local/" en "Example" "Example Printer, fully driverless, cups-filters 1.27.5" "MFG:Example;MDL:Printer;CMD:PWGRaster,AppleRaster,PWG,URF;"
DEBUG: PID 24180 (ippfind _ipps._tcp) exited with no errors.
DEBUG: PID 24184 (ippfind _ipp._tcp) exited with no errors.
```

### CODE REVIEW

File: utils/driverless.c [hunk 1]
```diff
@@ -46,7 +46,9 @@ static void		cancel_job(int sig);
 static int				
 compare_service_uri(char *a,	char *b)		
 {
-  size_t size = sizeof(a);
+  size_t size_a = strlen(a);
+  size_t size_b = strlen(b);
+  size_t size = (size_a>size_b ? size_b+1 :size_a +1);
   return (memcmp(a,b, size));
```
Comment (tillkamppeter) [inline utils/driverless.c:52]: Why not simply
```
compare_service_uri(char *a,	char *b)		
￼{
  return (strcmp(a, b));
}

**Comment by tillkamppeter** [comment]
The output of the first three commands is correct now, but please also check the output of
```
SOFTWARE=cups driverless
```
It should give exactly the same three items as
```
driverless
```
but in CUPS format and with the `DEBUG...` messages on stderr added.

**Comment by nidhijainpnp** [comment]
> The output of the first three commands is correct now, but please also check the output of
> 
> ```
> SOFTWARE=cups driverless
> ```
> 

> but in CUPS format and with the `DEBUG...` messages on stderr added.

Okay, I would update the code for mode = 2 also (SOTWARE =cups).

**Comment by nidhijainpnp** [comment]
Testing results:
```
nidhi@nidhi-X510UNR:~$ SOFTWARE=CUPS driverless
DEBUG: Started ippfind (PID 30623)
network ipps://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D._ipps._tcp.local/ "HP OfficeJet Pro 6960" "HP OfficeJet Pro 6960 (fully driverless)" "MFG:HP;MDL:OfficeJet Pro 6960;CMD:PCLM,PCL,PWGRaster,AppleRaster,JPEG,URF,PWG;" ""
network ipps://hello1._ipps._tcp.local/ "Example Printer" "Example Printer (fully driverless)" "MFG:Example;MDL:Printer;CMD:PWGRaster,AppleRaster,PWG,URF;" ""
DEBUG: Started ippfind (PID 30629)
network ipp://HP%20OfficeJet%20Pro%206960%20%5BC3B82D%5D%20(USB%201)._ipp._tcp.local/ "HP OfficeJet Pro 6960" "HP OfficeJet Pro 6960 (fully driverless)" "MFG:HP;MDL:OfficeJet Pro 6960;CMD:PCLM,PCL,PWGRaster,AppleRaster,JPEG,URF,PWG;" ""
DEBUG: PID 30623 (ippfind _ipps._tcp) exited with no errors.
DEBUG: PID 30629 (ippfind _ipp._tcp) exited with no errors.
```

**Comment by tillkamppeter** [comment]
Thank you very much. Now the HTPPS support is working as it should be.

**Pull request merged**